odoo/tools/freeze.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+
+# Minimalistic and odoo-specific version of https://github.com/spulec/freezegun
+# For license information, see https://www.apache.org/licenses/LICENSE-2.0.txt
+# Copyright 2012 Steve Pulec
+#
+# Modifications to the original code:
+#   * remove python 2 compatibility layer
+#   * restrict possible attributes to replace to a few names
+#   * remove module attribute cache because of the previous point
+#   * remove factories as "time moving" is not needed for Odoo (for now)
+#   * create fake functions/classes during initialization
+#   * cover only functions/methods used within Odoo
+#   * simplification of some heuristics
+#   * freeze_time only works as a class decorator for TestCase-based classes
+
+import inspect
+import functools
+import sys
+import datetime
+import time
+
+from unittest import TestCase
+
+
+def _get_module_attributes(module):
+    result = []
+    for attribute_name in dir(module):
+        try:
+            attribute_value = getattr(module, attribute_name)
+        except (ImportError, AttributeError, TypeError):
+            continue
+        else:
+            result.append((attribute_name, attribute_value))
+    return result
+
+
+class freeze_time:
+
+    """
+    Monkey-patches a series of time functions and classes from the CPython stdlib
+
+    The patched functions/classes are:
+        * datetime.datetime
+            * now
+            * today
+            * utcnow
+        * datetime.date
+            * today
+        * time.time
+        * time.strftime
+    """
+
+    def __init__(self, time_type):
```

Comment:
`time_type`?  What a confusing variable name!
```suggestion
    def __init__(self, t):
```